l10n_br_fiscal/models/subsequent_operation.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2020  KMEE
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+
+from __future__ import division, print_function, unicode_literals
+
+from odoo import api, fields, models, _
+from ..constants.fiscal import (
+    SITUACAO_EDOC,
+)
+
+
+SUBSEQUENT_SITUATION = (
```

Comment:
Este campo deveria ser substituído pelo document_type_id (l10n_br_fiscal.document.type) para informar qual documento deveria ser gerado nesta movimentação.

Reply:
> @renatonlima revisei seus comentários e também conversei com o @luismalta
> 
> O modelo SubSequentDocument é importante sim, quando implementamos isso até fizemos sem, mas ficou confuso e difícil de gerenciar.
> 
> O SUBSEQUENT_SITUATION é utilizado para definir quando o modelo será criado e não qual documento fiscal e série. A opção "remessa" ainda pode ser melhorada. Mas as outras estão corretas. Esse campo determina quando o documento subsequente deve ser criado:
> 
> * manualmente;
> * quando o documento original for autorizado;
> * quando o documento original for um cupom fiscal
> * etc.
> 
> Pedi pra ele tb adicionar um relacional com o tipo de documento fiscal e a série a ser aplicada, que pelo que eu entendi era isso o que vc estava querendo dizer.

Talvez seria interessante alterar o nome do campo para deixar mais explícito que este campo é como o documento subsequente será criado.

Reply:
Sobre o SubSequentDocument, eu acho quer seria interessante esse controle ficar no FiscalDocument sem ter a necessidade de outro objeto SubSequentDocument:

Dentro do FiscalDocument tendo os campos de documento de origem (source_document_id), e destino (subsequent_document_id) é possível visualizar e lista todos os documentos relacionados alterando o search_name por exemplo e também a sequencia de todos os documentos e é possível rastrear a sequencia, pelo que eu vi você só tem o document_subsequent_ids no documento original (primeiro documento);

É possível colocar dentro do form view do Documento Fiscal esses campo para visutalizar detalhadamente, também pode ter um campo calculado para visualizar toda cadeia de documentos equivalente ao document_subsequent_ids e os métodos _compute_document_subsequent_generated, _generates_subsequent_operations ;

Nos casos onde a criação do documento subsequente for manual, deve existir um botão no Fiscal Document para criar o documento subsequente.